### PR TITLE
Time agent objective code cleanup

### PR DESCRIPTION
**Pull request by ghost**
[tweak]
Changes the objects check in locate and rearrange objectives to use types instead of instantiated objects, also updates counting and list text output code to support passed in types.

### CODE REVIEW

File: code/__HELPERS/lists.dm [hunk 1]
```diff
@@ -367,10 +372,15 @@
...
+/proc/count_by_name(var/list/L, name)
 	var/i = 0
-	for(var/atom/T in L)
-		if(T.name == name)
+	var/atom/current_atom = null
+	var/current_name = ""
+	for(var/T in L)
+		if(isatom(T) || ispath(T))
+			current_atom = T
+		current_name = ispath(T) ? initial(current_atom.name) : isatom(T) ? current_atom.name : T
```
Comment (gurfan) [inline code/__HELPERS/lists.dm:382]: wouldn't this still cause a runtime when passed an item path?

Comment (ghost) [reply code/__HELPERS/lists.dm:382]: Didn't when I tested it

**Comment by Kurfursten** [comment]
Today I will remind him

**Comment by ghost** [comment]
Fixed the nagging issue with types being expanded into subtypes during is_type_in_list that caused me to leave it in draft so it's now fully tested working @Kurfursten

**Comment by Kurfursten** [comment]
You put no description in your PR body so I don't really know what I'm reviewing here.

**Comment by ghost** [comment]
> You put no description in your PR body so I don't really know what I'm reviewing here.

Just added one now.

**Pull request merged**